src/modules/attempts/attempts.service.ts

```diff
@@ -638,4 +639,322 @@ export class AttemptsService {
+                    };
+                  });
+                }
+                break;
+              
+              case QuestionType.MATCH:
+                if (userAnswerData.matches) {
+                  const matchAnswers = userAnswerData.matches;
+                  questionData.selectedOptions = matchAnswers.map((match: any) => {
+                    const option = question.options?.find(opt => opt.questionOptionId === match.optionId);
+                    const matchWithOption = question.options?.find(opt => opt.questionOptionId === match.matchWith);
+                    return {
+                      questionOptionId: match.optionId,
+                      text: option?.text || '',
+                      matchWith: match.matchWith,
+                      matchWithText: matchWithOption?.text || '',
+                    };
+                  });
+                }
+                break;
+              
+              case QuestionType.SUBJECTIVE:
+              case QuestionType.ESSAY:
+                if (userAnswerData.text) {
+                  questionData.selectedOptions = {text: userAnswerData.text};
+                }
+                break;
+            }
+          }
+
+          // Add correct answer information if showCorrectAnswer is enabled
+          if (test.showCorrectAnswer) {
+            switch (question.type) {
+              case QuestionType.MCQ:
+              case QuestionType.TRUE_FALSE:
+              case QuestionType.MULTIPLE_ANSWER: {
+                const correctOptions = question.options?.filter(opt => opt.isCorrect) || [];
+                questionData.correctOptions = correctOptions.map(opt => ({
+                  questionOptionId: opt.questionOptionId,
+                  text: opt.text,
+                  marks: opt.marks || 0,
+                }));
+                break;
+              }
+              
+              case QuestionType.FILL_BLANK: {
+                const correctBlanks = question.options?.filter(opt => opt.isCorrect) || [];
+                questionData.correctOptions = correctBlanks.map(opt => ({
+                  blankIndex: opt.blankIndex,
+                  text: opt.text,
+                  caseSensitive: opt.caseSensitive || false,
+                }));
+                break;
+              }
+              
+              case QuestionType.MATCH: {
+                const correctMatches = question.options?.filter(opt => opt.isCorrect) || [];
+                questionData.correctOptions = correctMatches.map(opt => ({
+                  questionOptionId: opt.questionOptionId,
+                  text: opt.text,
+                  matchWith: opt.matchWith,
+                }));
+                break;
+              }
+              
+              case QuestionType.SUBJECTIVE:
+              case QuestionType.ESSAY: {
+                const correctOptions = question.options?.filter(opt => opt.isCorrect) || [];
```

Comment:
correctOptions should only be send if the parameter on the test showCorrectAnswer is set to true.

Reply:
@vinayak-patil answerSheet should be sent only if answerSheet parameter at test is true